src/transports/sentry_transport_winhttp.c

```diff
@@ -51,10 +53,47 @@ sentry__winhttp_bgworker_state_free(void *_state)
     sentry__dsn_decref(state->dsn);
     sentry__rate_limiter_free(state->ratelimiter);
     sentry_free(state->user_agent);
+    sentry_free(state->proxy_username);
+    sentry_free(state->proxy_password);
     sentry_free(state->proxy);
     sentry_free(state);
 }
 
+// Function to extract and set credentials TODO replace with `sentry__url_parse`
```

Comment:
Since `sentry___url_parse` was built to handle DSNs, it won't parse the general `http_proxy` formats (e.g. the one we have in our test, `http://user:password@127.0.0.1:8080`, will fail the parsing as it has no `/path` behind the port). We would either need to add a flag to the parser signaling whether it needs the `/path` section or not, or we can keep this `set_proxy_credentials` function.

Reply:
I would add the flag as a parameter. We should let setting the proxy fail anyway after the parsing. If a proxy requires authentication, then it won't work on Windows. That should generate a `WARN` log.